### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by agenius-mohammed-ali**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,12 +6,118 @@ output:
...
+### Overview
+
+***periscope2*** is a scalable and UI-standardized 'shiny' framework including a variety of developer convenience functions with the goal of both streamlining robust application development and assisting in creating a consistent user experience regardless of application or developer.
+
+
+***periscope2*** is full of developer-friendly features such as:
+
+* A predefined but flexible templates for new Shiny applications with a default [bs4Dash](https://bs4dash.rinterface.com/) layout
```
Comment (cb4ds) [inline README.md:22]: Remove the "A", start with "Predefined"

File: README.md [hunk 2]
```diff
@@ -6,12 +6,118 @@ output:
...
+* Separation by file of functionality that exists in one of the three shiny scopes: global, server-global, and server-local
+* Generated applications are organized in an easy to follow and maintain folder structure based on files functionality
+* Off shelf and ready to be used modules ('Table Downloader', 'Plot Downloader', 'File Downloader' and 'Reset Application'
```
Comment (cb4ds) [inline README.md:25]: "Off shelf" should be "Off-the-shelf" (with dashes)

File: README.md [hunk 3]
```diff
@@ -6,12 +6,118 @@ output:
...
+  * More modules are introduced in each new version
```
Comment (cb4ds) [inline README.md:26]: remove this line

File: README.md [hunk 4]
```diff
@@ -6,12 +6,118 @@ output:
...
+
+#### Generated applications layout
+[periscope](https://cran.r-project.org/web/packages/periscope/index.html) full generated app layout is simple as follow:
+
+- Left sidebar
+- Right sidebar
+- Header
+- Body
+
+All above parts are customizable except for header, user can only update title part in it
```
Comment (cb4ds) [inline README.md:65]: remove this sentence and change "Header" to "Header (only can update title)" above

File: README.md [hunk 5]
```diff
@@ -6,12 +6,118 @@ output:
...
+
+However, periscope2 full generated app layout has the following structure:
```
Comment (cb4ds) [inline README.md:67]: [periscope2](link) full generated app layouts:"

File: README.md [hunk 6]
```diff
@@ -6,12 +6,118 @@ output:
...
+
+- Left sidebar
+- Right sidebar
+- Header
+- Footer
+- Body
+
+For periscope2, each part has it own UI R file that can customized and updated alone. Aslo, each part has its own dedicated alert panel except for header that has an additional alert panel for announcements framework.
```
Comment (cb4ds) [inline README.md:75]: "that can be customized and updated" to end the first sentence. 

Second sentence: "Each part also his its own dedicated alert panel, including an announcements framework as part of the header."

File: README.md [hunk 7]
```diff
@@ -6,12 +6,118 @@ output:
...
+
+-----
+
+#### Generated applications code structure
+
+- Regardless files contents, both packages generated applications share the same first level folder structure:
+
+<center><img src="./vignettes/figures/App_Structure.png"></center>
+
+- After that, there are differences in next level structure as follow:
```
Comment (cb4ds) [inline README.md:87]: change to "Differences in the next level of structure are:"

File: README.md [hunk 8]
```diff
@@ -6,12 +6,118 @@ output:
...
+  - **program** folder, that contains application main logic, periscope2 has:
+    - a dedicated UI R script for each layout part to control it
+    - **config**  - folder for storing application different configuration files (it will evolve more in next versions)
```
Comment (cb4ds) [inline README.md:90]: remove the part in parenthesis

File: README.md [hunk 9]
```diff
@@ -6,12 +6,118 @@ output:
...
+      - Initially contains announcements module configuration file
+    - **modules**  - folder for application shiny modules R scripts (it will evolve more in next versions)
```
Comment (cb4ds) [inline README.md:92]: remove the part in parenthesis

File: README.md [hunk 10]
```diff
@@ -6,12 +6,118 @@ output:
...
+    <center><img src="./vignettes/figures/Program_Folder.png"></center>
+  - ***www*** folder that contains the application JS, CSS and theme customization files
+    - Although generated applications by both frameworks will have the same folder structure, periscope2 generated applications will have two additional files **css/custom.css** and **js/custom.js** to have application custom JS and CSS code
```
Comment (cb4ds) [inline README.md:95]: I'd rework this simply as "**css/custom.css** and **js/custom.js** to hold custom application JS and CSS code"

File: README.md [hunk 11]
```diff
@@ -6,12 +6,118 @@ output:
...
+      - Note that both files are already linked to generated app and ready to be used immediately by user
```
Comment (cb4ds) [inline README.md:96]: change slightly to:
"- Both of these files are already linked into the generated app and ready to be used"

File: README.md [hunk 12]
```diff
@@ -6,12 +6,118 @@ output:
...
+
+----
+
+#### Packages modules
+
+- Both packages share some modules (for now) as downloadable file, downloadable plot and downloadable table
```
Comment (cb4ds) [inline README.md:102]: Both packages share the following modules: <list>

File: README.md [hunk 13]
```diff
@@ -6,12 +6,118 @@ output:
...
+- However, periscope2 has more modules (more to come with each new version) as Announcement module
```
Comment (cb4ds) [inline README.md:103]: Periscope2 has the following additional modules:
- Announcements

File: README.md [hunk 14]
```diff
@@ -6,12 +6,118 @@ output:
...
+
+While [periscope](https://cran.r-project.org/web/packages/periscope/index.html) modules can be called via shiny module old style (via `callModule` method) or using new shiny module style.  periscope2 modules can only be called using new shiny module style. Check [migrating to module server](https://shiny.posit.co/r/articles/improve/modules/#migrating-from-callmodule-to-moduleserver) article for more related technical info
+
+----
+
+#### fresh package Styling Support
+[fresh](https://dreamrs.github.io/fresh/) package is used in both packages to style generated applications different parts via configurations rather than code via **'www/periscope_style.yaml'** file.
+
+However, [shiny dashboard variables](https://dreamrs.github.io/fresh/articles/vars-shinydashboard.html) are used for styling [periscope](https://cran.r-project.org/web/packages/periscope/index.html) generated applications. While [bs4Dash related variables](https://dreamrs.github.io/fresh/articles/vars-bs4dash.html) are used for styling periscope2 generated applications but not via code only via **'www/periscope_style.yaml'** configuration file (hence the file key names are user friendly in both packages)
```
Comment (cb4ds) [inline README.md:118]: I'd work this sentence around the other way around and drop the periscope link and shinydashboard link:
periscope2 uses bs4dash related variables for styling in code and via the ... configuration file.  The file key names use shiny dashboard variables for user friendly styling in both packages.

**Review by cb4ds — CHANGES_REQUESTED** [review]
Looks Great, some tweaks to make

File: README.md (hunk 1 above)
Comment (agenius-mohammed-ali) [reply README.md:22]: done

File: README.md (hunk 2 above)
Comment (agenius-mohammed-ali) [reply README.md:25]: done

File: README.md (hunk 3 above)
Comment (agenius-mohammed-ali) [reply README.md]: done

File: README.md [hunk 15]
```diff
@@ -6,12 +6,118 @@ output:
...
+
+### periscope vs periscope2
+[periscope](https://cran.r-project.org/web/packages/periscope/index.html) is widely regarded and used an enterprise application development aid framework for providing the user with the ability to build a robust shiny applications and dashboards.
+
+[periscope](https://cran.r-project.org/web/packages/periscope/index.html) is the direct ancestor of periscope2 in which periscope2 took most of its basic philosophy and generated applications structure.
+
+However; as periscope2 is the next generation (iteration) of [periscope](https://cran.r-project.org/web/packages/periscope/index.html) package, they differ in many aspects as follow:
+
+#### Underlying frameworks
+[periscope](https://cran.r-project.org/web/packages/periscope/index.html) is depending on bootstrap 3 supported shiny packages:
```
[the remark this answers is not in the document]
Comment (agenius-mohammed-ali) [reply README.md:46]: done

File: README.md [hunk 16]
```diff
@@ -6,12 +6,118 @@ output:
...
+
+- [shinydashboard](https://cran.r-project.org/web/packages/shinydashboard/index.html)
+- [shinydashboard Plus](https://cran.r-project.org/web/packages/shinydashboardPlus/index.html)
+- [shinyBS](https://cran.r-project.org/web/packages/shinyBS/index.html)
+
+While periscope2 is depending only on bootstrap 4 supported package [bs4Dash](https://cran.r-project.org/web/packages/bs4Dash/index.html) to cover the same functions that are presented by [periscope](https://cran.r-project.org/web/packages/periscope/index.html) dependencies and more (i.e dashboard functionality, alerts, tool-tips, etc..) in more modernized look and feel
```
[the remark this answers is not in the document]
Comment (agenius-mohammed-ali) [reply README.md:52]: done

File: README.md [hunk 17]
```diff
@@ -6,12 +6,118 @@ output:
...
+
+----
+
+#### Generated applications layout
+[periscope](https://cran.r-project.org/web/packages/periscope/index.html) full generated app layout is simple as follow:
```
[the remark this answers is not in the document]
Comment (agenius-mohammed-ali) [reply README.md:57]: done

File: README.md (hunk 4 above)
Comment (agenius-mohammed-ali) [reply README.md:64]: done

File: README.md (hunk 5 above)
Comment (agenius-mohammed-ali) [reply README.md:66]: done

File: README.md (hunk 6 above)
Comment (agenius-mohammed-ali) [reply README.md:74]: done

File: README.md (hunk 7 above)
Comment (agenius-mohammed-ali) [reply README.md:86]: done

File: README.md [hunk 18]
```diff
@@ -6,12 +6,118 @@ output:
...
+-----
+
+#### Generated applications code structure
+
+- Regardless files contents, both packages generated applications share the same first level folder structure:
+
+<center><img src="./vignettes/figures/App_Structure.png"></center>
+
+- After that, there are differences in next level structure as follow:
+  - **program** folder, that contains application main logic, periscope2 has:
```
[the remark this answers is not in the document]
Comment (agenius-mohammed-ali) [reply README.md:87]: done

File: README.md [hunk 19]
```diff
@@ -6,12 +6,118 @@ output:
...
+    - a dedicated UI R script for each layout part to control it
```
[the remark this answers is not in the document]
Comment (agenius-mohammed-ali) [reply README.md:88]: done

File: README.md (hunk 8 above)
Comment (agenius-mohammed-ali) [reply README.md:89]: done

File: README.md (hunk 9 above)
Comment (agenius-mohammed-ali) [reply README.md:91]: done

File: README.md (hunk 10 above)
Comment (agenius-mohammed-ali) [reply README.md:94]: done

File: README.md (hunk 12 above)
Comment (agenius-mohammed-ali) [reply README.md:101]: done

File: README.md (hunk 13 above)
Comment (agenius-mohammed-ali) [reply README.md:102]: done

File: README.md [hunk 20]
```diff
@@ -6,12 +6,118 @@ output:
...
+    - Although generated applications by both frameworks will have the same folder structure, periscope2 generated applications will have two additional files **css/custom.css** and **js/custom.js** to have application custom JS and CSS code
+      - Note that both files are already linked to generated app and ready to be used immediately by user
+
+----
+
+#### Packages modules
+
+- Both packages share some modules (for now) as downloadable file, downloadable plot and downloadable table
+- However, periscope2 has more modules (more to come with each new version) as Announcement module
+- Also, there are modules that exist in both packages but behave a little differently than each other (i.e logger and Reset Application) modules
```
[the remark this answers is not in the document]
Comment (agenius-mohammed-ali) [reply README.md:103]: done

File: README.md [hunk 21]
```diff
@@ -6,12 +6,118 @@ output:
...
+----
+
+#### Old Shiny module style support
+
+While [periscope](https://cran.r-project.org/web/packages/periscope/index.html) modules can be called via shiny module old style (via `callModule` method) or using new shiny module style.  periscope2 modules can only be called using new shiny module style. Check [migrating to module server](https://shiny.posit.co/r/articles/improve/modules/#migrating-from-callmodule-to-moduleserver) article for more related technical info
+
+----
+
+#### fresh package Styling Support
+[fresh](https://dreamrs.github.io/fresh/) package is used in both packages to style generated applications different parts via configurations rather than code via **'www/periscope_style.yaml'** file.
```
[the remark this answers is not in the document]
Comment (agenius-mohammed-ali) [reply README.md:121]: done

File: README.md (hunk 14 above)
Comment (agenius-mohammed-ali) [reply README.md:123]: done

**Review by cb4ds — APPROVED** [review]

**Pull request merged**